algoliasearch-core/src/main/java/com/algolia/search/models/indexing/FiltersJsonDeserializer.java

```diff
@@ -63,4 +61,16 @@
                 })
             .collect(Collectors.toList());
   }
+
+  /** Build filters from (legacy) string */
+  private List<List<String>> buildFilters(String string) {
+    if (string.startsWith("(") && string.endsWith(")")) {
```

Comment:
Sorry for being nitpicky - I think we are missing on valid case, e.g.

```java
"(color:green,color:yellow),color:black" 
=> ["color:green,color:yellow","color:black"]
=> [["color:green","color:yellow"],["color:black"]]
```

^ can you confirm @godelized? :)

Reply:
`(color:green,color:yellow),color:black` means `(color = green OR color = yellow) AND color = black`.

It is equivalent to the following:

- `(color:green,color:yellow),(color:black)`
- `[["color:green","color:yellow"],["color:black"]]`
- `[["color:green","color:yellow"],"color:black"]`

Reply:
Is `(color:green,color:yellow),color:black` also a valid syntax? Are the parenthesis mandatory for the second group?

Reply:
Is it also possible to have nested parentheses, like `(color:green,(color:yellow,color:blue)),color:pink`? if yes, should we support it?

Reply:
> I think we should reject inner parenthesis in the Java client and fix the engine.

Since the legacy form can only come from the engine, and users can't provide such syntax, do we need the client to reject it?

Reply:
IMO it's the responsibility of the API to validate the payload, we shouldn't handle that case here. As @Aallam said, we'll retrieve rules already saved, meaning that the filters were validated by the API when saving the rule beforehand.

Reply:
Makes sense but how do you deal with `(((A:1),B:2),(C:3),D:4)`?

Reply:
Maybe I am missing something - but in this case we have inner parenthesis `(A:1)` and hence this filter shouldn't be validated by the API, right?

Reply:
If I understand correctly we convert the filter string to the JSON array format: https://github.com/algolia/algoliasearch-client-java-2/pull/771/files#diff-9135e60921839756742dc454f4b063cefc2f1e574560ca6ea51adb77ca28460bR35-R38.
Did I miss something?
So I am afraid that we'd convert `(((A:1),B:2),(C:3),D:4)` to something wrong.